cpp/src/arrow/flight/types.cc

```diff
@@ -83,6 +84,31 @@ Status FlightInfo::GetSchema(std::shared_ptr<Schema>* out) const {
   return Status::OK();
 }
 
+Location::Location() {
+  uri_ = std::shared_ptr<arrow::internal::Uri>(new arrow::internal::Uri());
```

Comment:
Using [std::make_shared](https://en.cppreference.com/w/cpp/memory/shared_ptr/make_shared) is generally the preferred idiom.